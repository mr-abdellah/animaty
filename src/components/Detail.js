/* eslint-disable max-len */
import { useParams } from 'react-router-dom';
import { useSelector } from 'react-redux';
import '../styles/detail.css';

const Detail = () => {
  const animes = useSelector((state) => state.animes.animes);
  const { title } = useParams();
  return (
    <>
      {animes?.filter((anime) => anime.title === title)
        .map((anime) => (
          <div className="anime-detail-container" key={anime.id}>
            <img src={anime.img} alt={anime.title} />
            <h3 className="anime-detail-title">{anime.title}</h3>
            <p className="anime-detail-description">
              <span>Description:</span>
              <br />
              {anime.description}
            </p>
            <p className="anime-detail-year">{`Release Date: ${anime.year}`}</p>
            <p className="anime-detail-type">{`Type : ${anime.type}`}</p>
            <p className="anime-detail-episodes">{`Episodes: ${anime.episodes}`}</p>
            <p className="anime-detail-rating">{`Allowed Just For: ${anime.rating}`}</p>
          </div>
        ))}
    </>
  );
};

export default Detail;
