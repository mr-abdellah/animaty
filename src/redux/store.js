import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import animeReducer from './cards';

const store = configureStore({
  reducer: {
    animes: animeReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
