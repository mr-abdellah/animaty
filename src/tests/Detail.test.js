import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Detail from '../components/Detail';

it('renders properly', () => {
  const tree = renderer.create(
    <Router>
      <Provider store={store}>
        <Detail />
      </Provider>
    </Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
