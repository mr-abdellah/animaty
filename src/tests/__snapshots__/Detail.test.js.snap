// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Detail page renders properly 1`] = `null`;
